frontend/packages/pipelines-plugin/src/components/import/pipeline/PipelineTemplate.tsx

```diff
@@ -76,17 +77,48 @@ const PipelineTemplate: React.FC<PipelineTemplateProps> = ({ builderImages, exis
       }
 
       const imagePipelines: PipelineKind[] = pipelineStorageRef.current[image.selected] || [];
-      const resourceSpecificPipeline = imagePipelines.find(
+      const resourceSpecificPipelines = imagePipelines.filter(
         (pl) => pl.metadata?.labels?.[labelType] === resources,
       );
-      const pipelineTemplate =
-        resourceSpecificPipeline || imagePipelines.find((pl) => !pl.metadata?.labels?.[labelType]);
+      const candidatePipelines = [
+        ...resourceSpecificPipelines,
+        ...imagePipelines.filter((pl) => !pl.metadata?.labels?.[labelType]),
+      ];
+
+      if (
+        pipelineTemplates
+          .map((pl) => pl.metadata?.name)
+          .sort()
+          .join(',') !==
+        candidatePipelines
+          .map((pl) => pl.metadata?.name)
+          .sort()
+          .join(',')
+      ) {
+        setPipelineTemplates(candidatePipelines);
+      }
```

Comment:
```suggestion
    const getPipelineNames = (pipelines: PipelineKind[]): string =>
             pipelines
          .map((pl) => pl.metadata?.name)
          .sort()
          .join(',');  

      if (getPipelineNames(pipelineTemplates) !== getPipelineNames(candidatePipelines)) {
        setPipelineTemplates(candidatePipelines);
      }
      
```

Maybe move `getPipelineNames` helper function out of the useEffect ?

Reply:
thank you, updated